components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2444,12 +2446,15 @@ public void createNewAPIVersion(API api, String newVersion) throws DuplicateAPIE
             String resourcePath = APIUtil.getOpenAPIDefinitionFilePath(api.getId().getApiName(),
                     api.getId().getVersion(), api.getId().getProviderName());
             if (registry.resourceExists(resourcePath + APIConstants.API_OAS_DEFINITION_RESOURCE_NAME)) {
+                String apiDefinition = definitionFromOpenAPISpec.getAPIDefinition(api.getId(), registry);
+                LinkedHashMap map = new ObjectMapper().readValue(apiDefinition, LinkedHashMap.class);
```

Comment:
Below change to `version` attribute should be done on this `map` object. Not the `swaggerObject`. `swaggerObject` is not needed now.